kube/src/client/body.rs

```diff
@@ -0,0 +1,32 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::stream::Stream;
+use http_body::Body;
+use pin_project::pin_project;
+
+// Wrap `http_body::Body` to implement `Stream`.
+#[pin_project]
+pub struct IntoStream<B> {
+    #[pin]
+    body: B,
+}
+
+impl<B> IntoStream<B> {
+    pub(crate) fn new(body: B) -> Self {
+        Self { body }
+    }
+}
+
+impl<B> Stream for IntoStream<B>
+where
+    B: Body,
```

Comment:
This seems reasonable, pretty light weight. It looks like there were even thoughts of doing this in `http-body` at some point: https://github.com/hyperium/http-body/pull/36